examples/logger.py

```diff
@@ -133,6 +136,29 @@ def __init__(self, name: str, parent: Optional[QObject] = None):
         logger = logging.getLogger()
         logger.setLevel(logging.DEBUG)
         logger.addHandler(self.handler)
+        self.initLevelSelector(self.loggerFrame.levelSelector)
+
+    def initLevelSelector(self, box: QComboBox):
+        """Initializes a combo box for selecting level."""
+        box.addItem("DEBUG")
+        box.addItem("INFO")
+        box.addItem("WARNING")
+        box.addItem("ERROR")
+        box.addItem("CRITICAL")
+        box.activated[str].connect(self.fixLevel)
+
+    def fixLevel(self, text):
+        """Responds to the levelSelector widget and changes the handler's level."""
+        if text == "DEBUG":
+            self.handler.setLevel(logging.DEBUG)
+        if text == "INFO":
+            self.handler.setLevel(logging.INFO)
+        if text == "WARNING":
+            self.handler.setLevel(logging.WARNING)
+        if text == "ERROR":
+            self.handler.setLevel(logging.ERROR)
+        if text == "CRITICAL":
+            self.handler.setLevel(logging.CRITICAL)
```

Comment:
1. `text` can be any string and `fixLevel()` is a public method. In this case, you should notify that "nothing happens when an invalid `text` is given". Or, if you can guarantee that invalid inputs are never given, you can change it to an internal (private) method `_fixLevel()`.
2. Use `elif` for efficiency.

In this case, the code is simple enough so might not be a problem, but there are several other approaches for this kind of task (you may keep the current code):
- Use [`enum`](https://docs.python.org/3/library/enum.html)
- Use [`getattr()`](https://docs.python.org/3/library/functions.html#getattr)
- Use a dictionary:
```python
level = {
    "DEBUG": logging.DEBUG,
    "INFO": logging.INFO,
    ...
}[text]
```

Reply:
I will use dictionary and set function as public.